pkg/common/sign/sign.go

```diff
@@ -0,0 +1,108 @@
+package sign
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	"github.com/square/sharkey/pkg/server/config"
+	"github.com/square/sharkey/pkg/server/storage"
+	"golang.org/x/crypto/ssh"
+)
+
+type Signer struct {
+	signer  ssh.Signer
+	conf    *config.Config
+	storage storage.Storage
+}
+
+func NewSigner(signer ssh.Signer, conf *config.Config, storage storage.Storage) Signer {
+	return Signer{
+		signer:  signer,
+		conf:    conf,
+		storage: storage,
+	}
+}
+
+func (s *Signer) Sign(keyId string, principals []string, certType uint32, pubkey ssh.PublicKey, extensions map[string]string) (*ssh.Certificate, error) {
```

Comment:
since now this lives in its own package we could maybe simplify the API and have `SignHostCert` and `SignUserCert` separated (can still be the same underlying private function)?